### DISCUSSION TITLE
Support modern auth methods to connect Keycloak app to database server

### DISCUSSION THREAD

**Discussion by tspearconquest** [keycloak/keycloak#21965] [Ideas]
What sort of support (or +1's for a feature request) is there for modern authentication methods to connect the Keycloak app to its database?

Context: When undergoing an audit for SOC2 compliance, an issue was highlighted that our services should not be using DB roles and passwords for connecting to the DB, and that we should instead make use of federated authentication for this purpose.

Currently, we have a Postgres DB server (Azure DB for Postgres Flexible Server) configured for Keycloak and we create a dedicated role for the keycloak service and appropriate privileges to prevent abuse; such as a dedicated schema, and least privileges required for operation of the Keycloak service at the database, schema, and table levels.

We are looking to switch to using Azure AD for the authentication of the Keycloak app to the database services, and as part of that switch, we would like to start to make use of [Managed Identities](https://github.com/MicrosoftDocs/azure-dev-docs/blob/main/articles/java/spring-framework/migrate-postgresql-to-passwordless-connection.md)

I realize the above link refers to a Java Spring app; I had tried to find docs around Quarkus apps, but I was only able to find various docs which detail how to accomplish this without a code change, by making use of Azure Container Instances and the Service Connector. Unfortunately, it's really not very flexible because we run our Keycloak instance on Kubernetes and would need to migrate the instance from Kubernetes to an Azure Container instance, which we hope to avoid for a few different reasons.

Primarily, we would like to make use of Azure Workload Identity, which is offered in the cluster, and is considered probably the most secure way (IMHO) to make use of managed identities across all of our apps which are behind Keycloak, in addition to Keycloak itself. But with that being said, I'm interested to see who else would be interested in Keycloak offering support for various cloud provider authentication methods (be it Azure cloud or someone else like Google cloud or AWS)

**Comment by Daveography** [comment]
@tspearconquest While it would be great if this was integrated, here are some instructions on how I got it to work with the current version of Keycloak: https://github.com/keycloak/keycloak/discussions/26267#discussioncomment-10824627

**Comment by Speeddymon** [comment]
Thanks, unfortunately I'm not working for that org anymore and they were in the process of moving from Keycloak to Okta. That said, it would be great if Keycloak team could take your implementation details and built a way for the Keycloak build/config process to make use of it so that it can be enabled as easily as possible.
